api/envoy/api/v2/route/route.proto

```diff
@@ -291,33 +291,52 @@ message RouteMatch {
   // is true.
   google.protobuf.BoolValue case_sensitive = 4;
 
-  oneof runtime_specifier {
-    // Indicates that the route should additionally match on a runtime key. An integer between
-    // 0-100. Every time the route is considered for a match, a random number between 0-99 is
-    // selected. If the number is <= the value found in the key (checked first) or, if the key is
-    // not present, the default value, the route is a match (assuming everything also about the
-    // route matches). A runtime route configuration can be used to roll out route changes in a
-    // gradual manner without full code/config deploys. Refer to the :ref:`traffic shifting
-    // <config_http_conn_man_route_table_traffic_splitting_shift>` docs for additional
-    // documentation.
-    //
-    // .. attention::
-    //
-    //   **This field is deprecated**. Set the
-    //   :ref:`runtime_fraction<envoy_api_field_route.RouteMatch.runtime_fraction>` field instead.
-    core.RuntimeUInt32 runtime = 5 [deprecated = true];
-
-    // Indicates that the route should additionally match on a runtime key. Every time the route
-    // is considered for a match, it must also fall under the percentage of matches indicated by
-    // this field. For some fraction N/D, a random number in the range [0,D) is selected. If the
-    // number is <= the value of the numberator N, or if the key is not present, the default
-    // value, the router continues to evaluate the remaining match criteria. A runtime_fraction
-    // route configuration can be used to roll out route changes in a gradual manner (with more
-    // granularity than the deprecated runtime field) without full code/config deploys. Refer to
-    // the :ref:`traffic shifting <config_http_conn_man_route_table_traffic_splitting_shift>` docs
-    // for additional documentation.
-    core.RuntimeFractionalPercent runtime_fraction = 9;
```

Comment:
I wasn't around (or wasn't watching) for the original conversation between yourself and @htuch on this, but I did have one thought about how to keep this structure but also fix the perf problems. What if we build this directly into runtime itself? Much like how we try to detect integers during runtime load, what if we also tried to detect a fractional percent and load it? Then, there could be a version of featureEnabled() that looks for an already loaded fractional percent or uses a default? This might get us the best of both worlds and be generally useful for the future? Thoughts?

Reply:
I like that a lot. It never occurred to me to do something like that because of my limited understanding of how `getInteger()` worked, but after poking around it makes sense to do it this way.

I'll send out another patch.

Reply:
+1 that sounds like the best option, thanks.